vcd/resource_vcd_nsxt_alb_service_engine_group_test.go

```diff
@@ -0,0 +1,273 @@
+data "vcd_nsxt_alb_importable_cloud" "cld" {
+  name          = "{{.ImportableCloud}}"
+  controller_id = local.controller_id
+}
+
+# Local variable is used to avoid direct reference and cover Terraform core bug https://github.com/hashicorp/terraform/issues/29484
+# Even changing NSX-T ALB Controller name in UI, plan will cause to recreate all resources depending 
+# on vcd_nsxt_alb_importable_cloud data source if this indirect reference (via local) variable is not used.
+locals {
+  controller_id = vcd_nsxt_alb_controller.first.id
+}
+
+resource "vcd_nsxt_alb_controller" "first" {
+  name         = "{{.ControllerName}}"
+  description  = "first alb controller"
+  url          = "{{.ControllerUrl}}"
+  username     = "{{.ControllerUsername}}"
+  password     = "{{.ControllerPassword}}"
+  license_type = "ENTERPRISE"
+}
+
+resource "vcd_nsxt_alb_cloud" "first" {
+  name        = "nsxt-cloud"
+  description = "first alb cloud"
+
+  controller_id       = vcd_nsxt_alb_controller.first.id
+  importable_cloud_id = data.vcd_nsxt_alb_importable_cloud.cld.id
+  network_pool_id     = data.vcd_nsxt_alb_importable_cloud.cld.network_pool_id
+}
+`
+
+const testAccVcdNsxtAlbServiceEngineStep1 = testAccVcdNsxtAlbServiceEnginePrereqs + `
+resource "vcd_nsxt_alb_service_engine_group" "first" {
+  name                      = "first-se"
+  alb_cloud_id              = vcd_nsxt_alb_cloud.first.id
+  service_engine_group_name = "Default-Group"
+  reservation_model         = "DEDICATED"
+}
+`
+
+const testAccVcdNsxtAlbServiceEngineStep2 = testAccVcdNsxtAlbServiceEnginePrereqs + `
+resource "vcd_nsxt_alb_service_engine_group" "first" {
+  name                      = "first-se-updated"
+  description               = "test-description"
+  alb_cloud_id              = vcd_nsxt_alb_cloud.first.id
+  service_engine_group_name = "Default-Group"
+  reservation_model         = "SHARED"
+}
+`
+
+const testAccVcdNsxtAlbServiceEngineStep3DS = testAccVcdNsxtAlbServiceEnginePrereqs + `
+# skip-binary-test: Data Source test
+resource "vcd_nsxt_alb_service_engine_group" "first" {
+  name                      = "first-se-updated"
+  description               = "test-description"
+  alb_cloud_id              = vcd_nsxt_alb_cloud.first.id
+  service_engine_group_name = "Default-Group"
+  reservation_model         = "SHARED"
+}
+
+data "vcd_nsxt_alb_service_engine_group" "first" {
+  name = vcd_nsxt_alb_service_engine_group.first.name
+}
+`
+
+const testAccVcdNsxtAlbServiceEngineStep4 = testAccVcdNsxtAlbServiceEnginePrereqs + `
+resource "vcd_nsxt_alb_service_engine_group" "first" {
+  name                      = "first-se-updated"
+  description               = "test-description"
+  alb_cloud_id              = vcd_nsxt_alb_cloud.first.id
+  service_engine_group_name = "Default-Group"
+  reservation_model         = "SHARED"
+  
+  # This feature remains not fully tested as it will impact some of the attributes, but only when tenant operations
```

Comment:
Nit. Should we add "TODO" to not forget such things? :)
```suggestion
  # TODO: This feature remains not fully tested as it will impact some of the attributes, but only when tenant operations
```

Reply:
Fixed